### PR TITLE
move signal ignore out of `eth/common`

### PR DESCRIPTION
**Pull request by arnetheduck**
`net/nat` is the only place where it is used and it certainly doesn't belong in `common`

### CODE REVIEW

File: eth/net/nat.nim [hunk 1]
```diff
@@ -47,6 +47,37 @@ var
...
+# handling elsewhere.
+proc ignoreSignalsInThread*() =
+  when defined(posix):
+    var signalMask, oldSignalMask: Sigset
+
+    # sigprocmask() doesn't work on macOS, for multithreaded programs
+    if sigfillset(signalMask) != 0:
+      echo osErrorMsg(osLastError())
+      quit(QuitFailure)
+    when defined(boehmgc):
```
Comment (tersec) [inline eth/net/nat.nim:63]: It's fine to keep the code as is to start with, but is the boehm GC something worth supporting in general with this code? Most of it by bulk is a special case for a memory manager not in use

**Review by tersec — APPROVED** [review]

**Comment by tersec** [comment]
```
 [NimScript] exec: nim c  --styleCheck:usages --styleCheck:error --verbosity:0 --hints:off --skipUserCfg --nimcache:build/nimcache -f --warning[ObservableStores]:off -d:nimOldCaseObjects -d:chronicles_log_level=TRACE --threads:on -d:release  --mm:refc -r --outdir:build/p2p tests/p2p/all_tests
/Users/runner/work/nim-eth/nim-eth/eth/p2p/rlpx.nim(972, 1) template/generic instantiation of `p2pProtocol` from here
/Users/runner/work/nim-eth/nim-eth/eth/p2p/rlpx.nim(978, 14) template/generic instantiation of `append` from here
/Users/runner/work/nim-eth/nim-eth/eth/rlp/writer.nim(311, 15) template/generic instantiation of `appendImpl` from here
/Users/runner/work/nim-eth/nim-eth/eth/rlp/writer.nim(159, 11) template/generic instantiation of `append` from here
/Users/runner/work/nim-eth/nim-eth/eth/rlp/writer.nim(311, 15) template/generic instantiation of `appendImpl` from here
/Users/runner/work/nim-eth/nim-eth/eth/rlp/writer.nim(296, 7) template/generic instantiation of `appendRecordType` from here
/Users/runner/work/nim-eth/nim-eth/eth/rlp/writer.nim(293, 21) template/generic instantiation of `enumerateRlpFields` from here
/Users/runner/work/nim-eth/nim-eth/eth/rlp/object_serialization.nim(17, 9) template/generic instantiation of `op` from here
/Users/runner/work/nim-eth/nim-eth/eth/rlp/writer.nim(291, 13) template/generic instantiation of `append` from here
/Users/runner/work/nim-eth/nim-eth/eth/rlp/writer.nim(308, 16) Warning: Signed integers cannot reliably be encoded using RLP [User]
/Users/runner/work/nim-eth/nim-eth/eth/p2p/peer_pool.nim(112, 57) template/generic instantiation of `async` from here
/Users/runner/work/nim-eth/nim-eth/eth/p2p/peer_pool.nim(152, 12) template/generic instantiation of `setResult` from here
/Users/runner/work/nim-eth/nim-eth/eth/p2p/peer_pool.nim(133, 25) Error: type mismatch
Expression: inc(rlpx_connect_success)
  [1] rlpx_connect_success: typedesc[IgnoredCollector]

Expected one of (first mismatch at [position]):
[1] func inc(a: var StInt; w: Word = 1)
[1] func inc(a: var StUint; w: Word = 1)
[1] proc inc(address: var TransportAddress; v: int = 1)
[1] proc inc[A](t: CountTableRef[A]; key: A; val = 1)
[1] proc inc[A](t: var CountTable[A]; key: A; val = 1)
[1] proc inc[T, V: Ordinal](x: var T; y: V = 1)
```
which is presumably because the import/export network dropped `metrics` here but it was never the right place to pull it in

File: eth/net/nat.nim (hunk 1 above)
Comment (arnetheduck) [reply eth/net/nat.nim]: nah, boehm is dead, might as well get rid of it here and now

**Review by kdeme — APPROVED** [review]

**Comment by arnetheduck** [comment]
> 

if peer_pool wants metrics, it should import metrics itself

turns out it's using public gauges declared in: [eth/p2p/rlpx.nim](https://github.com/status-im/nim-eth/pull/734/files#diff-c40f379771d62d8ac66047d2e3434df3f634c49009f23aaa39d903fea6624fea)

**Pull request merged**